File: server/group-service/src/main/java/vn/edu/tdtu/dto/response/Notification.java
package vn.edu.tdtu.dto.response;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import vn.tdtu.common.enums.notification.ENotificationType;

import java.util.Date;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class Notification {
    private String userFullName;
    private String avatarUrl;
    private String content;
    private String refId;
    private String title;
    private String fromUserId;
    private List<String> toUserIds;
    private ENotificationType type;
    private Date createAt;
}

File: web-client/src/types/notification.ts
import { ENotificationType } from "./enum";
import { User } from "./user";

export type InteractNotification = {
    id: string;
    content: string;
    title: string;
    refId: string;
    type: ENotificationType;
    createAt: string;
    fromUser: User;
    hasRead: boolean;
    toUserId: string;
};

export type MessageNotification = {
    id: string;
    createdAt: string;
    avatar: string;
    content: string;
    imageUrls: string[];
    fromUserId: string;
    toUserId: string;
    title: string;
}


export type NotificationSocketMessage = {
    userFullName: string;
    avatarUrl: string;
    content: string;
    refId: string;
    title: string;
    fromUserId: string;
    toUserIds: string[];
    type: ENotificationType;
    createAt: string;
}

export type Violation = {
    id: string;
    refId: string;
    content: string;
}